go/vt/tabletserver/tabletserver.go

```diff
@@ -184,6 +188,9 @@ func (tsv *TabletServer) SetQueryRules(ruleSource string, qrs *QueryRules) error
 
 // GetState returns the name of the current TabletServer state.
 func (tsv *TabletServer) GetState() string {
+	if tsv.lameduck.Get() != 0 {
+		return "NOT_SERVING"
```

Comment:
If you look at my last change, you'll see that I initially had it as LAMEDUCK. I had also renamed one of the states to NOT_CONNECTED. But some tests started failing. So, I thought it was better not to be too granular.
Is there a benefit to someone knowing that a server is in lameduck mode?
Basically, I'm going by the precedent that other google servers don't reveal this state.